tests/core/channels/test_slack.py

```diff
@@ -0,0 +1,798 @@
+@pytest.mark.asyncio
+async def test_slackbot_send_custom_json():
+    from rasa.core.channels.slack import SlackBot
+
+    with aioresponses() as mocked:
+        mocked.post(
+            "https://www.slack.com/api/chat.postMessage",
+            payload={"ok": True, "purpose": "Testing bots"},
+        )
+
+        bot = SlackBot("DummyToken", "General")
+        await bot.send_custom_json("ID", {"test_key": "test_value"})
+
+        r = latest_request(mocked, "POST", "https://www.slack.com/api/chat.postMessage")
+
+        assert r
+
+        request_params = json_of_latest_request(r)
+
+        assert request_params == {
+            "as_user": True,
+            "channel": "General",
+            "test_key": "test_value",
+        }
+
+
+@pytest.mark.filterwarnings("ignore:unclosed.*:ResourceWarning")
+@pytest.mark.asyncio
+async def test_slackbot_send_custom_json_threaded():
+    from rasa.core.channels.slack import SlackBot
+
+    with aioresponses() as mocked:
+        mocked.post(
+            "https://www.slack.com/api/chat.postMessage",
+            payload={"ok": True, "purpose": "Testing bots"},
+        )
+
+        bot = SlackBot("DummyToken", "General", thread_id="DummyThread")
+        await bot.send_custom_json("ID", {"test_key": "test_value"})
+
+        r = latest_request(mocked, "POST", "https://www.slack.com/api/chat.postMessage")
+
+        assert r
+
+        request_params = json_of_latest_request(r)
+
+        assert request_params == {
+            "as_user": True,
+            "channel": "General",
+            "thread_ts": "DummyThread",
+            "test_key": "test_value",
+        }
+
+
+def prepare_slack_request(headers):
+    request = Request(
+        b"/webhooks/slack/webhook",
+        headers=Header(headers),
+        version="1.1",
+        method="POST",
+        transport=None,
+        app=None,
+    )
+    request.body = b"""{"foo": "bar"}"""
+    return request
+
+
+def test_slack_verify_signature_is_always_true_if_there_is_no_key():
+    request = prepare_slack_request(
+        {
+            "x-slack-signature": "foobar",  # this is an invalid signature
+            "x-slack-request-timestamp": "1604586653",
+        }
+    )
+    slack = SlackInput("mytoken", slack_signing_secret=None)
+
+    assert slack.is_request_from_slack_authentic(request) is True
+
+
+def test_slack_verify_signature():
+    request = prepare_slack_request(
+        {
+            "x-slack-signature": "v0=80a3bd62ce5af04d8d80781134f165df"
+            "185b90342d467abf5c74a27d2d0dd1f5",
+            "x-slack-request-timestamp": "1604586653",
+        }
+    )
+    input_with_right_secret = SlackInput("mytoken", slack_signing_secret="foobar")
+    input_with_wrong_secret = SlackInput("mytoken", slack_signing_secret="foobaz")
+
+    assert input_with_right_secret.is_request_from_slack_authentic(request) is True
+
+    assert input_with_wrong_secret.is_request_from_slack_authentic(request) is False
```

Comment:
optional: in my opinion this should be two different unit tests as or we should use `pytest.mark.parametrize` (maybe merge with the next test then as well)